UnitySDK/Assets/ML-Agents/Scripts/Timer.cs

```diff
@@ -0,0 +1,257 @@
+using UnityEngine;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine.Profiling;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+#if UNITY_EDITOR
+using UnityEditor;
+
+#endif
+
+
+namespace MLAgents
+{
+
+    [DataContract]
```

Comment:
I used the generic C# DataContract/DataMember for JSON serialization. Could also use https://docs.unity3d.com/Manual/JSONSerialization.html - this happens very rarely so I'm not too worried about performance.

Reply:
I would lean towards using Unity classes.

Reply:
Tried this as a quick proof of concept. I don't think the Unity serializer can handle recursively defined classes, even if they don't loop in practice.

I made an output class (to help match the format of python's output):
```csharp
    [Serializable]
    public class TimerOutput
    {
        public string name = "";
        public double total;
        public double self;
        public int num_calls;
        List<TimerOutput> children;

        public TimerOutput(TimerNode node)
        {
            total = node.TotalSeconds;
            self = node.SelfSeconds;
            num_calls = node.m_NumCalls;
            // note children not assigned
        }
    }
```
and used it with 
```csharp
var output = new TimerOutput(m_RootNode);
var json = JsonUtility.ToJson(output, prettyPrint: true);
```
and get this
> Serialization depth limit 7 exceeded at 'MLAgents::TimerOutput.children'. There may be an object composition cycle in one or more of your serialized classes.
and it doesn't output the `children` list.

Reply:
oof